### PR TITLE
Add a new filter to dataset page, to filter by “Formats”, “Years” , “…

### LINKED ISSUE #141: Add a new filter to dataset page, to filter by “Formats”, “Years” , “Geography”, “License”
As a data steward , I want to be able to filter the main page of datasets search results by the following fields:
- Formats (e.g CSV, TXT, JSON, XML)
- Years
- Geography (e.g. the location where this datasets applies to describes)
- License

Such that, it's easy for me to location datasets for my work , when I'm interested in certain years, places , formats of data and licenses to do my work.

## Tasks

- [ ] Find the existing logic, possibly from search facets , to do the filtering.
- [ ] Refactor , or create a new snippet for the dataset filter bar.
- [ ] Create the fields needed for the filtering if not already existing.
- [ ] Test

## Analysis

In CKAN we already have a way to filter data according to tags, publishers (offices, tags, categories etc) it would be wise to try to re use this logic instead of writing new one, but we may need to write new logic if that existing supports has some sorts of special affinity to the data attributes it already applies to (office, tages, categories etc).

### PR DESCRIPTION
**Pull request by sivang**
…Geography”, “License” , closes #141

Does not include styling!

### CODE REVIEW

File: ckanext/ed/plugin.py [hunk 1]
```diff
@@ -137,4 +137,8 @@ def dataset_facets(self, facets_dict, package_type):
         facets_dict['groups'] = "Major Topics"
         facets_dict['tags'] = "Tags"
         facets_dict['organization'] = "Publishers"
+        facets_dict['res_format'] = "Formats"
+        facets_dict['spatial'] = "Geography"
+        facets_dict['license_id'] = "License"
+        facets_dict['start_date'] = "Years"
```
Comment (zelima) [inline ckanext/ed/plugin.py]: Think "Years" is a bit strange name for the filter. How about "Year", or maybe "Date"

File: ckanext/ed/schemas/dataset.json [hunk 2]
```diff
@@ -191,16 +201,6 @@
...
-    {
-      "field_name": "Start Date",
-      "label": "Data Relevance Start Date",
-      "preset": "date"
-    },
-    {
-      "field_name": "End Date",
-      "lable": "Data Relevance End Date",
-      "preset": "date"
-    },
```
Comment (zelima) [inline ckanext/ed/schemas/dataset.json]: What this change is about, Just moved up? any reasons?

Comment (sivang) [reply ckanext/ed/schemas/dataset.json]: moved back to dataset page for the 'Years' filter

File: ckanext/ed/plugin.py (hunk 1 above)
Comment (sivang) [reply ckanext/ed/plugin.py]: no, this is the name Ivan specified in design

**Pull request merged**